flags/14.0/vttablet.txt

```diff
@@ -195,32 +100,34 @@ Usage of vttablet:
       --enable_hot_row_protection                                        If true, incoming transactions for the same row (range) will be queued and cannot consume all txpool slots.
       --enable_hot_row_protection_dry_run                                If true, hot row protection is not enforced but logs if transactions would have been queued.
       --enable_lag_throttler                                             If true, vttablet will run a throttler service, and will implicitly enable heartbeats
-      --enable_query_plan_field_caching                                  This option fetches & caches fields (columns) when storing query plans (default true)
+      --enable_query_plan_field_caching                                  (DEPRECATED) This option fetches & caches fields (columns) when storing query plans (default true)
       --enable_replication_reporter                                      Use polling to track replication lag.
-      --enable_semi_sync                                                 Enable semi-sync when configuring replication, on primary and replica tablets only (rdonly tablets will not ack).
```

Comment:
`--enable_semi_sync` is deprecated using pflag's `MarkDeprecated' and hence doesn't show up in the usage text. Just want to confirm that it is what we have intended.

There are other flags which are marked using (DEPRECATED) in the help text, like `--enable_query_plan_field_caching`, so they show up in the usage. Do we want to resolve this inconsistency  now or wait since they will be removed soon anyway?

Reply:
Ideally we should use `MarkDeprecated` for all deprecated flags.

Reply:
The decision is that we will mark all Deprecated flags as `MarkDeprecated`.